mne/source_estimate.py

```diff
@@ -1098,6 +1097,81 @@ def transform(self, func, idx=None, tmin=None, tmax=None, copy=False):
+        nanoAmperes (i.e., MNE-like solutions, *not* dSPM or sLORETA).
+
+        Reference:
+        Goldenholz, D. M., Ahlfors, S. P., Hämäläinen, M. S., Sharon, D.,
+        Ishitobi, M., Vaina, L. M., & Stufflebeam, S. M. (2009).
+        Mapping the Signal-To-Noise-Ratios of Cortical Sources in
+        Magnetoencephalography and Electroencephalography.
+        Human Brain Mapping, 30(4), 1077–1086. doi:10.1002/hbm.20571
+
+        Parameters
+        ----------
+        info : instance of io.meas_info.Info
+            The measurement info.
+        fwd : instance of Forward
+            The forward solution used to create the source estimate.
+        cov : instance of Covariance
+            The noise covariance used to estimate the resting cortical
+            activations. Should be an evoked covariance, not empty room.
+
+        Returns
+        -------
+        snr_stc : instance of SourceEstimate
+            The source estimate with the SNR computed.
+
+        Notes
+        -----
+        We define the SNR in decibels for each source location at each
+        time point as:
+
+        .. math::
+
+            {\\rm SNR} = 10\\log_10[\\frac{a^2}{N}\\sum_k\\frac{b_k^2}{s_k^2}]
+
+        where :math:`\\b_k` is the signal on sensor :math:`k` provided by the
+        forward model for a source with unit amplitude, :math:`a` is the
+        source amplitude, :math:`N` is the number of sensors, and
+        :math:`s_k^2` is the noise variance on sensor :math:`k`.
+        """
+        from .forward import convert_forward_solution
+        from .minimum_norm.inverse import _prepare_forward
+        from mne import SourceEstimate
+        
+        M = np.shape(self.data)[0]
+        T = np.shape(self.data)[1]
+        snr_stc = SourceEstimate(np.zeros((M, T)), self.vertices, self.tmin, 
+                                 self.tstep, self.subject)
+        
+        fwd = convert_forward_solution(fwd, surf_ori=True, force_fixed=True)
+        
+        # G is gain matrix [ch x src], cov is noise covariance [ch x ch]
+        _, _, G, _, _, _, _, cov, _ = _prepare_forward(fwd, info, cov,
+                                                                fixed=True, 
+                                                                loose=0, 
+                                                                rank=None, 
+                                                                pca=False,
+                                                                use_cps=True, 
+                                                                exp=None, 
+                                                                limit_depth_chs=False, 
+                                                                combine_xyz='fro',
+                                                                allow_fixed_depth=True, 
+                                                                limit=None)
+
+        N = cov['dim'] # number of sensors/channels    
```

Comment:
we call it `n_channels` elsewhere

Reply:
Should I look up N from n_channels instead of the way I did it?

Reply:
I used this because the dim of the cov matrix was a little smaller than the n_chan you get from info (this was the case in the example that I attached.)

Reply:
yes that's fine, I was just suggesting that the variable name that you assign it to should be called `n_channels`.

Reply:
Ok will do that now.